Documentation/ApiOverview/Backend/ContextualMenu.rst

```diff
@@ -312,3 +312,22 @@ If :yaml:`autoconfigure` is disabled you can manually register a context menu it
 by adding the tag :yaml:`backend.contextmenu.itemprovider`:
 
 ..  include:: /CodeSnippets/Tutorials/ContextMenu/ManualServicesYaml.rst.txt
+
+Migration of the JavaScript
+===========================
+
+..  deprecated:: 12.0
+    Due to historical reasons, a context menu item was bound to
+    :js:`this` in its callback action which was used to access the
+    context menu item's :js:`dataset`. With TYPO3 v12.0 the invocation of the
+    assigned callback actions is adapted to pass the :js:`dataset` as the
+    third argument.
+
+    Binding the context menu item to :js:`this` in the callback is marked as
+    deprecated.
+
+To access data attributes, use the :js:`dataset` argument passed as the third
+argument in the context menu callback action.
+
+.. literalinclude:: _dataset-version-11.js
```

Comment:
The snippet is not rendered, first it is in "_ContextMenu/_dataset-version-11.js" . but changing does not help, maybe the ":diff:" thingy?

Reply:
It works with the correct paths, but the result is not so nice, so I changed it to two different code-snippets